### PR TITLE
[feature](datetime) "timediff" supports calculating microseconds.

### PR DESCRIPTION
**Pull request by Mryange**
## Proposed changes

Because the original local environment has been deleted, the current PR is pick https://github.com/apache/doris/pull/21371. 
In the previous code, we distinguished between two types of time, "time" and "timev2", but on the backend, both of them are the same.
So here, we are making some modifications. The unit for "time" is seconds, while the unit for "timev2" is microseconds.
Issue Number:  https://github.com/apache/doris/issues/17174
Due to the mixed usage of "time" and "timev2" in many places in the past, this pull request will be **converted to a draft** first for regression testing.

before
```
mysql [test]>select timediff(
cast('0001-01-01 00:00:00.114514' as Datetimev2(6) ) , 
cast('0001-01-01 00:00:00.000000' as Datetimev2(6) ));
+--------------------------------------------------------------------------------------------------------------------+
| timediff(CAST('0001-01-01 00:00:00.114514' AS DATETIMEV2(6)), CAST('0001-01-01 00:00:00.000000' AS DATETIMEV2(6))) |
+--------------------------------------------------------------------------------------------------------------------+
| 00:00:00                                                                                                   |
+--------------------------------------------------------------------------------------------------------------------+
```
after
```
mysql [test]>select timediff(
cast('0001-01-01 00:00:00.114514' as Datetimev2(6) ) , 
cast('0001-01-01 00:00:00.000000' as Datetimev2(6) ));
+--------------------------------------------------------------------------------------------------------------------+
| timediff(CAST('0001-01-01 00:00:00.114514' AS DATETIMEV2(6)), CAST('0001-01-01 00:00:00.000000' AS DATETIMEV2(6))) |
+--------------------------------------------------------------------------------------------------------------------+
| 00:00:00.114514                                                                                                    |
+--------------------------------------------------------------------------------------------------------------------+
```

<--Describe your changes.-->

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run beut

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run arm

File: gensrc/script/doris_builtins_functions.py [hunk 1]
```diff
@@ -1036,13 +1036,13 @@
     [['timediff'], 'TIMEV2', ['DATETIMEV2', 'DATETIMEV2'], ''],
 
     [['datediff'], 'INT', ['DATETIMEV2', 'DATEV2'], ''],
-    [['timediff'], 'TIMEV2', ['DATETIMEV2', 'DATEV2'], ''],
+    [['timediff'], 'TIME', ['DATETIMEV2', 'DATEV2'], ''],
```
Comment (morrySnow) [inline gensrc/script/doris_builtins_functions.py:1039]: why use time for datetimev2 and datev2? do not need microsecond diff?

Comment (Mryange) [reply gensrc/script/doris_builtins_functions.py:1039]: I think that timev2 should only be calculated when both types have microseconds.

Comment (Mryange) [reply gensrc/script/doris_builtins_functions.py:1039]: If want to calculate microseconds,  cast "datev2" to "datetimev2"?

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

File: gensrc/script/doris_builtins_functions.py (hunk 1 above)
Comment (morrySnow) [reply gensrc/script/doris_builtins_functions.py]: if the diff include second, it should include microsecond too

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p0

File: be/src/vec/data_types/data_type_time.h [hunk 2]
```diff
@@ -69,6 +69,34 @@ class DataTypeTime final : public DataTypeNumberBase<Float64> {
 
     DataTypeSerDeSPtr get_serde() const override { return std::make_shared<DataTypeTimeSerDe>(); };
     TypeIndex get_type_id() const override { return TypeIndex::Time; }
+    const char* get_family_name() const override { return "time"; }
 };
 
+class DataTypeTimeV2 : public DataTypeNumberBase<Float64> {
```
Comment (Gabriel39) [inline be/src/vec/data_types/data_type_time.h:75]: need to implement `to_pb_column_meta`

Comment (Gabriel39) [reply be/src/vec/data_types/data_type_time.h:75]: also need to hold a `scale` inside this type

File: gensrc/script/doris_builtins_functions.py [hunk 3]
```diff
@@ -1048,13 +1048,13 @@
         [['timediff'], 'TIMEV2', ['DATETIMEV2', 'DATETIMEV2'], ''],
 
         [['datediff'], 'INT', ['DATETIMEV2', 'DATEV2'], ''],
-        [['timediff'], 'TIMEV2', ['DATETIMEV2', 'DATEV2'], ''],
```
Comment (Gabriel39) [inline gensrc/script/doris_builtins_functions.py:1051]: why make this change?

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run buildall

**Review by morrySnow — DISMISSED** [review]

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 39.37 seconds
 stream load tsv:          455 seconds loaded 74807831229 Bytes, about 156 MB/s
 stream load json:         21 seconds loaded 2358488459 Bytes, about 107 MB/s
 stream load orc:          58 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          29 seconds loaded 861443392 Bytes, about 28 MB/s
 insert into select:          69.9 seconds inserted 10000000 Rows, about 143K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230622023342_clickbench_pr_166483.html

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run clickbench

**Comment by Mryange** [comment]
run clickbench

**Comment by Mryange** [comment]
run clickbench

**Comment by Mryange** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 44.27 seconds
 stream load tsv:          454 seconds loaded 74807831229 Bytes, about 157 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          57 seconds loaded 1101869774 Bytes, about 18 MB/s
 stream load parquet:          28 seconds loaded 861443392 Bytes, about 29 MB/s
 insert into select:          70.1 seconds inserted 10000000 Rows, about 142K ops/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230623095434_clickbench_pr_166677.html

**Pull request closed without merging**